src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -2373,6 +2376,192 @@ public void canSearchByPurchaseOrderLineIdentifierProperty() throws Exception {
+
+    final UUID statisticalCodeId = UUID.fromString(
+      statisticalCode.getJson().getString("id")
+    );
+
+    final String status = "Available";
+
+    final JsonObject itemToCreate = new ItemRequestBuilder()
+      .forHolding(holdingsRecordId)
+      .withMaterialType(journalMaterialTypeId)
+      .withPermanentLoanType(canCirculateLoanTypeId)
+      .withStatus(status)
+      .withStatisticalCodeIds(Arrays.asList(statisticalCodeId))
+      .create();
+
+    final Response createdItem = itemsClient.attemptToCreate(itemToCreate);
+
+    assertThat(createdItem.getStatusCode(), is(HttpURLConnection.HTTP_CREATED));
+  }
+
+  @Test
+  public void cannotCreateItemWithNonExistentStatisticalCodeId() throws Exception {
+    final UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+    final UUID nonExistentStatisticalCodeId = UUID.randomUUID();
+    final String status = "Available";
+
+    final JsonObject itemToCreate = new ItemRequestBuilder()
+      .forHolding(holdingsRecordId)
+      .withMaterialType(journalMaterialTypeId)
+      .withPermanentLoanType(canCirculateLoanTypeId)
+      .withStatus(status)
+      .withStatisticalCodeIds(Arrays.asList(nonExistentStatisticalCodeId))
+      .create();
+
+    final Response createdItem = itemsClient.attemptToCreate(itemToCreate);
+
+    assertThat(createdItem, hasValidationError(
+      "Statistical code does not exist", "statisticalCodeIds",
+      nonExistentStatisticalCodeId.toString()));
+  }
+
+  @Test
+  public void canCreateItemWithMultipleStatisticalCodeIds() throws Exception {
+    final var firstStatisticalCode = statisticalCodeFixture
+      .createSerialManagementCode(new StatisticalCodeBuilder()
+        .withCode("stcone")
+        .withName("Statistical code 1"));
+
+    final var secondStatisticalCode = statisticalCodeFixture
+      .attemptCreateSerialManagementCode(new StatisticalCodeBuilder()
+        .withCode("stctwo")
+        .withName("Statistical code 2"));
+
+    final UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    final UUID firstStatisticalCodeId = UUID.fromString(
+      firstStatisticalCode.getJson().getString("id")
+    );
+    final UUID secondStatisticalCodeId = UUID.fromString(
+      secondStatisticalCode.getJson().getString("id")
+    );
+
+    final String status = "Available";
+
+    final JsonObject itemToCreate = new ItemRequestBuilder()
+      .forHolding(holdingsRecordId)
+      .withMaterialType(journalMaterialTypeId)
+      .withPermanentLoanType(canCirculateLoanTypeId)
+      .withStatus(status)
+      .withStatisticalCodeIds(Arrays.asList(
+        firstStatisticalCodeId,
+        secondStatisticalCodeId
+      ))
+      .create();
+
+    final Response createdItem = itemsClient.attemptToCreate(itemToCreate);
+
+    assertThat(createdItem.getStatusCode(), is(HttpURLConnection.HTTP_CREATED));
+  }
+
+  @Test
+  public void cannotCreateItemWithAllNonExistentStatisticalCodeId() throws Exception {
```

Comment:
What about the scenario where some of the statistical codes exist and others do not.

Reply:
That is this test.

Reply:
The title suggests to me that all of the statistical codes don't exist. Have I misunderstood the use of _all_ in the name?

Reply:
Sure, I can update name.